src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -0,0 +1,388 @@
+import net.minecraft.nbt.NBTTagList;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import java.util.function.Predicate;
+
+import static com.minecolonies.api.util.constant.Suppression.DEPRECATION;
+
+/**
+ * Utility methods for the inventories.
+ */
+public final class ItemStackUtils
+{
+    /**
+     * Variable representing the empty itemstack in 1.10.
+     * Used for easy updating to 1.11
+     */
+    public static final ItemStack EMPTY = ItemStack.EMPTY;
+
+    /**
+     * Predicate to check if an itemStack is empty.
+     */
+    @NotNull
+    public static final Predicate<ItemStack> EMPTY_PREDICATE = ItemStackUtils::isItemStackEmpty;
+
+    /**
+     * Negation of the itemStack empty predicate (not empty).
+     */
+    @NotNull
+    public static final Predicate<ItemStack> NOT_EMPTY_PREDICATE = EMPTY_PREDICATE.negate();
+
+    /**
+     * The compound id for fortune enchantment.
+     */
+    private static final int FORTUNE_ENCHANT_ID = 35;
+
+    /**
+     * Private constructor to hide the implicit one.
+     */
+    private ItemStackUtils()
+    {
+        /*
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Verifies if there is one tool with an acceptable level
+     * in a worker's inventory.
+     *
+     * @param stack        the stack to test.
+     * @param toolType     the type of tool needed
+     * @param minimalLevel the minimum level for the tool to find.
+     * @param maximumLevel the maximum level for the tool to find.
+     * @return true if tool is acceptable
+     */
+    public static boolean hasToolLevel(@Nullable final ItemStack stack, final IToolType toolType, final int minimalLevel, final int maximumLevel)
+    {
+        if (isItemStackEmpty(stack))
+        {
+            return false;
+        }
+
+        final int level = getMiningLevel(stack, toolType);
+        return isTool(stack, toolType) && verifyToolLevel(stack, level, minimalLevel, maximumLevel);
+    }
+
+    /**
+     * Wrapper method to check if a stack is empty.
+     * Used for easy updating to 1.11.
+     *
+     * @param stack The stack to check.
+     * @return True when the stack is empty, false when not.
+     */
+    @NotNull
+    public static Boolean isItemStackEmpty(@Nullable final ItemStack stack)
+    {
+        return stack == null || stack == EMPTY || stack.getCount() <= 0;
+    }
+
+    /**
+     * Checks if this ItemStack can be used as a Tool of type.
+     *
+     * @param itemStack Item to check.
+     * @param toolType  Type of the tool.
+     * @return true if item can be used, otherwise false.
+     */
+    public static boolean isTool(@Nullable final ItemStack itemStack, final IToolType toolType)
+    {
+        if (isItemStackEmpty(itemStack))
+        {
+            return false;
+        }
+
+        boolean isATool = false;
+        if (ToolType.AXE.equals(toolType) || ToolType.SHOVEL.equals(toolType) || ToolType.PICKAXE.equals(toolType))
```

Comment:
Having a second look I got confused, it seems like in the case of axe, pickaxe, shovel we do check the mining level.
But in the case of the other tools we just make sure that the ToolType we receive is equal to the itemStack.